### PR TITLE
Remove redundant token BWC tests

### PR DESCRIPTION
**Pull request by n1v0lg**
This PR removes YAML cluster upgrade tests for token backwards
compatibility. The Java REST test suite `TokenBackwardsCompatibilityIT`
fully covers all scenarios included in the YAML tests already. Due to
the complex test setup, and since this particular YAML test suite is
not meant to be tested with different clients, Java REST tests are more
appropriate. 

The test suite we are removing is resource intensive, both in terms of
build times and maintenance, and as such not justified as a redundant
suite.

Relates: #77350

### CODE REVIEW

File: x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/mixed_cluster/50_token_auth.yml [hunk 1]
```diff
@@ -1,180 +0,0 @@
----
-"Get the indexed token and use if to authenticate":
```
Comment (n1v0lg) [inline x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/mixed_cluster/50_token_auth.yml:2]: Covered by: https://github.com/elastic/elasticsearch/blob/main/x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/TokenBackwardsCompatibilityIT.java#L57

File: x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/old_cluster/50_token_auth.yml [hunk 2]
```diff
@@ -1,130 +0,0 @@
----
-"Create a token and reuse it across the upgrade":
```
Comment (n1v0lg) [inline x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/old_cluster/50_token_auth.yml:2]: Covered by: https://github.com/elastic/elasticsearch/blob/main/x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/TokenBackwardsCompatibilityIT.java#L80

File: x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/old_cluster/50_token_auth.yml [hunk 3]
```diff
@@ -1,130 +0,0 @@
...
-
-  - match: { type: "Bearer" }
-  - is_true: access_token
-  - set: { access_token: refreshed_access_token }
-  - is_true: refresh_token
-  - set: { refresh_token: refreshed_refresh_token }
-  - match: { expires_in: 3600 }
-  - is_false: scope
-
-  # test refresh token (use it)
```
Comment (n1v0lg) [inline x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/old_cluster/50_token_auth.yml:104]: Covered by: https://github.com/elastic/elasticsearch/blob/main/x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/TokenBackwardsCompatibilityIT.java#L80

File: x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/mixed_cluster/50_token_auth.yml (hunk 1 above)
Comment (n1v0lg) [inline x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/mixed_cluster/50_token_auth.yml:2]: Covered by: https://github.com/elastic/elasticsearch/blob/main/x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/TokenBackwardsCompatibilityIT.java#L126

File: x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/mixed_cluster/50_token_auth.yml [hunk 4]
```diff
@@ -1,180 +0,0 @@
...
-      headers:
-        Authorization: Bearer ${token}
-      search:
-        rest_total_hits_as_int: true
-        index: token_index
-
-  - match: { hits.total: 8 }
-
----
-"Get the indexed refreshed access token and use if to authenticate":
```
Comment (n1v0lg) [inline x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/mixed_cluster/50_token_auth.yml:58]: Covered by: https://github.com/elastic/elasticsearch/blob/main/x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/TokenBackwardsCompatibilityIT.java#L126

(Token at idx 4)

File: x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/mixed_cluster/50_token_auth.yml [hunk 5]
```diff
@@ -1,180 +0,0 @@
...
-      headers:
-        Authorization: Bearer ${token}
-      search:
-        rest_total_hits_as_int: true
-        index: token_index
-
-  - match: { hits.total: 8 }
-
----
-"Get the indexed refresh token and use it to get another access token and authenticate":
```
Comment (n1v0lg) [inline x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/mixed_cluster/50_token_auth.yml:109]: Covered by: https://github.com/elastic/elasticsearch/blob/main/x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/TokenBackwardsCompatibilityIT.java#L169

In the above Java REST test, we "verify new nodes can refresh tokens created by old nodes and vice versa" so the scenario of using an old token on a new node is covered

File: x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/upgraded_cluster/50_token_auth.yml [hunk 6]
```diff
@@ -1,92 +0,0 @@
----
-"Get the indexed token and use if to authenticate":
```
Comment (n1v0lg) [inline x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/upgraded_cluster/50_token_auth.yml:2]: Covered by: https://github.com/elastic/elasticsearch/blob/main/x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/TokenBackwardsCompatibilityIT.java#L214

File: x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/upgraded_cluster/50_token_auth.yml [hunk 7]
```diff
@@ -1,92 +0,0 @@
...
-  - do:
-      headers:
-        Authorization: Bearer boom
-      catch: /unable to authenticate with provided credentials and anonymous access is not allowed for this request/
-      search:
-        rest_total_hits_as_int: true
-        index: token_index
-
----
-"Get the indexed refresh token and use if to get another access token and authenticate":
```
Comment (n1v0lg) [inline x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/upgraded_cluster/50_token_auth.yml:48]: Covered by: https://github.com/elastic/elasticsearch/blob/main/x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/TokenBackwardsCompatibilityIT.java#L232

**Comment by n1v0lg** [comment]
I will follow this PR up with an actual fix addressing #77350. 

For the current PR, I just wanted another pair of eyes to confirm we are not losing coverage.

**Review by jakelandis — APPROVED** [review]
LGTM, thanks for the links to the tests with duplication

**Review by albertzaharovits — APPROVED** [review]
LGTM
I appreciate the effort to track down all these!

**Pull request merged**